### ISSUE TITLE
Add threshold at a rule level so large apps can adopt this incrementally

### ISSUE THREAD

**Issue by MiguelMadero** [ember-template-lint/ember-template-lint#144]
With large applications, it can be hard to adopt good rules all at once and not doing that means that we'll keep adding new offenders. An option to prevent this is to set a threshold or an exact count of "allowed occurrences" in order to prevent new ones from popping up until we get a chance to fix the existing ones. 

The configuration could look something like

```
module.exports = {
  rules: {
    'bare-strings': { allowedOccurrences: 44 }
  }
}
```

cc: @zen-tmq @rwjblue

**Comment by rwjblue** [comment]
FWIW, we already have a per rule & per template "pending" concept. I definitely agree that we should add a threshold to this though. 

Would that work for y'all?

**Comment by MiguelMadero** [comment]
Im not sure if threshold or exact number of allowed occurrences. If someone
fixes an instance, they should lower that number, otherwise someone else
could add a new one later of we arent forced to decrease that number.

**Comment by rwjblue** [comment]
Sounds good. Then the only thing we need to do is update the pending concept to support `count` property I think?

**Comment by MiguelMadero** [comment]
The original [PR to add pending](https://github.com/rwjblue/ember-template-lint/pull/22/files) can be a good start. I see that pending is per-file, however I think it would make sense to have a count per-rule. Currently: 

```
 rules: {...},
 pending: ['module', 'anotherModule']
```

The simplest implementation for all of this is to have a single count for all rules. Since it's an exact match, there is less need to make a distinction about which rules are failing...

```
 rules: {...},
 pending: 1983
```

... However, that opens the possibility for someone fixing one and adding a new on the same commit without realizing it. For that reason, It might make more sense to do it at the rule level, instead, maybe a count or an object with a count, or maybe a combination. 

```
 rules: {'some-rule': true, 'another-rule': 22, 'yet-another-rule: {allowedCount: 42}},
 pending: ['module', 'anotherModule', {'many-rule': 42}, {'even-more': ['module1', {module2: 4}]}]
```

tl;dr; my preference

```
 rules: {'some-rule': true},
 pending: [{'some-rule': 42}, 'module1'] // everything in my module and a total of 42 for some-rule (inclussive)
```

I think the last one is more explicit without getting super complicated and hard to maintain for both the addon author and consumers of the addon. 

Sorry for belaboring too much on this, but since it would introduce a new config option, I want to make sure you're happy with it before we push forward. Also we need to consider it doesn't conflict with https://github.com/rwjblue/ember-template-lint/issues/136, I think at least the latter option is fine since it doesn't rely on `extends` or `config` provided by addons since all the exceptions are configured inside of `pending`

**Comment by bertdeblock** [comment]
Is this ticket still relevant now that ember-template-lint has the concept of TODO's?

**Comment by locks** [comment]
I'm closing the issue because I think the answer is that the ticket is indeed no longe relevant!
